src/Adapter/MSTestAdapter.PlatformServices/Services/ReflectionOperations2.cs

```diff
@@ -23,26 +26,26 @@ public ReflectionOperations2()
 #pragma warning disable IL2026 // Members attributed with RequiresUnreferencedCode may break when trimming
 #pragma warning disable IL2067 // 'target parameter' argument does not satisfy 'DynamicallyAccessedMembersAttribute' in call to 'target method'.
 #pragma warning disable IL2057 // Unrecognized value passed to the typeName parameter of 'System.Type.GetType(String)'
-    public IEnumerable<ConstructorInfo> GetDeclaredConstructors(Type classType)
-        => classType.GetTypeInfo().DeclaredConstructors;
+    public ConstructorInfo[] GetDeclaredConstructors(Type classType)
+        => classType.GetConstructors(DeclaredOnlyLookup);
 
     public MethodInfo? GetDeclaredMethod(Type type, string methodName)
-        => type.GetTypeInfo().GetDeclaredMethod(methodName);
+        => type.GetMethod(methodName, DeclaredOnlyLookup);
 
-    public IEnumerable<MethodInfo> GetDeclaredMethods(Type classType)
-        => classType.GetTypeInfo().DeclaredMethods;
+    public MethodInfo[] GetDeclaredMethods(Type classType)
+        => classType.GetMethods(DeclaredOnlyLookup);
 
-    public IEnumerable<PropertyInfo> GetDeclaredProperties(Type type)
-        => type.GetTypeInfo().DeclaredProperties;
+    public PropertyInfo[] GetDeclaredProperties(Type type)
+        => type.GetProperties(DeclaredOnlyLookup);
 
     public PropertyInfo? GetDeclaredProperty(Type type, string propertyName)
-        => type.GetTypeInfo().GetDeclaredProperty(propertyName);
+        => type.GetProperty(propertyName, DeclaredOnlyLookup);
 
     public Type[] GetDefinedTypes(Assembly assembly)
-        => assembly.DefinedTypes.ToArray();
+        => assembly.GetTypes();
```

Comment:
`DefinedTypes` alone would internally allocate two arrays, and then we do `ToArray` again which will allocate a third array.

`GetTypes()` seems to be much more efficient. I assume it doesn't introduce any behavior changes.